pkg/db/advisory_locks.go

```diff
@@ -0,0 +1,198 @@
+package db
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"hash/fnv"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/openshift-online/rh-trex/pkg/logger"
+	"gorm.io/gorm"
+)
+
+type (
+	advisoryLockMap map[string]*AdvisoryLock
+	LockType        string
+)
+
+const (
+	Migrations LockType = "migrations"
+	Dinosaurs  LockType = "dinosaurs"
+)
+
+type LockFactory interface {
+	NewAdvisoryLock(ctx context.Context, id string, lockType LockType) (string, error)
```

Comment:
nit: worth adding some document for each method.

Reply:
added